src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -15,6 +15,9 @@ namespace System.Text.RegularExpressions
     /// RegexCompiler translates a block of RegexCode to MSIL, and creates a
     /// subclass of the RegexRunner type.
     /// </summary>
+    [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2060:MakeGenericMethod",
+        Target = "M:System.Text.RegularExpressions.RegexCompiler.#cctor",
+        Justification = "The referenced methods don't have any DynamicallyAccessedMembers annotations. See https://github.com/mono/linker/issues/1727")]
```

Comment:
Unrelated, are all of the patterns employed below recognized by the linker? i.e. it's able to see it must preserve the IsBoundary method, the runtrack field, the char.IsDigit method, etc?

Reply:
The ILLinker isn't warning for these methods:

```C#
        private static FieldInfo RegexRunnerField(string fieldname) => typeof(RegexRunner).GetField(fieldname, BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Instance | BindingFlags.Static)!;

        private static MethodInfo RegexRunnerMethod(string methname) => typeof(RegexRunner).GetMethod(methname, BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Instance | BindingFlags.Static)!;
```

My understanding is because it sees the specified `typeof(RegexRunner)` and thinks "I must preserve all Methods and Fields on the RegexRunner Type, since these methods try getting fields and methods on this Type".

But @vitek-karas and @MichalStrehovsky would know for sure if the above reasoning is correct.

Reply:
I can validate by debugging the linker if needed, but @eerhardt is right - the call to `GetField` will fallback to basically mark all fields which fulfill the binding flags, if it can't figure out the name of the field statically (like in this case). Same goes for `GetMethod`.

Reply:
Thanks for the info.